spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -237,51 +206,50 @@ def __init__(self, ipyclient, additional_options, interpreter_versions,
         # Show a message in our installers to explain users how to use
         # modules that don't come with them.
         self.show_modules_message = is_pynsist() or running_in_mac_app()
-        self.shutdown_lock = Lock()
 
     # ---- Public API ---------------------------------------------------------
-    def shutdown_kernel(self):
-        """Shutdown kernel."""
-        with self.shutdown_lock:
-            # Avoid calling shutdown_kernel on the same manager twice
-            # from different threads to avoid crash.
-            if self.kernel_manager.shutting_down:
-                return
-            self.kernel_manager.shutting_down = True
-        try:
-            self.kernel_manager.shutdown_kernel()
-        except Exception:
-            # kernel was externally killed
-            pass
+    @property
+    def is_spyder_kernel(self):
+        if self.kernel_handler is None:
+            return False
+        return self.kernel_handler.known_spyder_kernel
+
+    def connect_kernel(self, kernel):
+        """Connect to kernel."""
+        # Kernel client
+        kernel_client = kernel.kernel_client
+        kernel_client.stopped_channels.connect(self.notify_deleted)
+        kernel_client.start_channels()
+        self.kernel_client = kernel_client
+
+        self.kernel_manager = kernel.kernel_manager
+        self.kernel_handler = kernel
+        # Send message to kernel to check status
+        self.check_spyder_kernel()
+        self.sig_shellwidget_created.emit(self)
```

Comment:
```suggestion
    def connect_kernel(self, kernel_handler):
        """Connect to the kernel using our handler."""
        # Kernel client
        kernel_client = kernel_handler.kernel_client
        kernel_client.stopped_channels.connect(self.notify_deleted)
        kernel_client.start_channels()
        self.kernel_client = kernel_client

        self.kernel_manager = kernel_handler.kernel_manager
        self.kernel_handler = kernel_handler
        
        # Send message to kernel to check status
        self.check_spyder_kernel()
        self.sig_shellwidget_created.emit(self)
```

To clarify that we're using our handler on this method.